scripts/hooks/customer-notes-guard.py

```
@@ -88,30 +109,220 @@ def _mutation_command_touches_customer_notes(command: str) -> bool:
+# sister hooks stay aligned).
+_OPEN_WRITE_RE = re.compile(
+    r"""open\(\s*['"]([^'"]+)['"]\s*,\s*['"]([^'"]*[waxA+][^'"]*)['"]"""
+)
+
+# Kwarg form: ``open('path', mode='w')`` / ``open('path', buffering=0, mode='wb')``.
+_OPEN_WRITE_KWARG_RE = re.compile(
+    r"""open\(\s*['"]([^'"]+)['"][^)]*\bmode\s*=\s*['"]([^'"]*[waxA+][^'"]*)['"]"""
+)
+
+
+def _extract_open_write_paths(body: str):
```

Ruff (0.15.12)

[warning] Missing return type annotation for private function `_extract_open_write_paths`

(ANN202)